### DISCUSSION TITLE
Creating integration cmake files for non-cmake projects ?

### DISCUSSION THREAD

**Discussion by Tryum** [microsoft/vcpkg#20743] [Q&A]
Would creating some custom cmake files for integrating non-cmake libraries be a good idea ?

I'm currently using freeimage vcpkg port and it's really painful to integrate so I'm wondering if it would make sense to create cmake files to ease installation of such libraries.

**Comment by jasjuang** [comment]
The best way is do a PR for the original project with the cmake files. If for some reason the original owner of the repo is not willing to merge the change then yes you can do it on vcpkg too. For example, you can look at the leveldb port in vcpkg.

**Comment by vicroms** [comment]
Multiple ports already do that, examples are: gtk, lua, osi, tbb, gl3w, glib, pthreads, etc.

I think an example of how to port a library that lacks CMake support by shipping a custom CMakeLists.txt would be useful.

I'll leave the issue open until such an example is given in the docs.
